doc/source/io.rst

```diff
@@ -440,6 +440,59 @@ individual columns:
     Specifying ``dtype`` with ``engine`` other than 'c' raises a
     ``ValueError``.
 
+.. note::
+
+   Reading in data with mixed dtypes and relying on ``pandas``
+   to infer them is not recommended. In doing so, the parsing engine will
```

Comment:
I think the sentence is little confusing. Maybe you're referring to a case mixed dtype contained a single columns? (I think inferring different dtypes each column should work well...)